test/e2e/framework/fixture.go

```diff
@@ -71,13 +71,17 @@ func (f *KCPFixture) SetUp(t *testing.T) func() {
 
 		// Wait for the server to become ready
 		go func(s *kcpServer) {
+			defer wg.Done()
 			err := s.Ready()
 			require.NoErrorf(t, err, "kcp server %s never became ready: %v", s.name, err)
-			wg.Done()
 		}(srv)
 	}
 	wg.Wait()
```

Comment:
Ready() will look 60s for the admin kubeconfig. Can we move that logic into the Run func and terminate the test immediately when the server dies?